contribs/gmf/src/directives/layertree.js

```diff
@@ -204,8 +207,69 @@ gmf.LayertreeController = function($http, $sce, $scope, ngeoCreatePopup,
   function() {
     this.map.render();
   }.bind(this));
+
+  // watch any change on dimensions object to refresh the layers
+  $scope.$watchCollection(function() {
+    return this.dimensions;
+  }.bind(this), function() {
+    this.updateDimensions_(this.gmfTreeManager_.tree);
+  }.bind(this));
+
+};
+
+
+/**
+ * @param {GmfThemesNode} node Layer tree node.
+ * @private
+ */
+gmf.LayertreeController.prototype.updateDimensions_ = function(node) {
+  node.children.forEach(function(childNode) {
+    if (childNode.children) {
+      this.updateDimensions_(childNode);
+    } else if (childNode.dimensions) {
+      var layersArray = this.map.getLayerGroup().getLayersArray();
+      var layer = this.layerHelper_.getLayerByName(childNode.name, layersArray);
+      if (layer) {
+        goog.asserts.assertInstanceof(layer, ol.layer.Layer);
+        this.updateLayerDimensions_(layer, childNode);
+      }
+    }
+  }, this);
 };
 
+
+/**
+ * @param {ol.layer.Layer} layer Layer to update.
+ * @param {GmfThemesNode} node Layer tree node.
+ * @private
+ */
+gmf.LayertreeController.prototype.updateLayerDimensions_ = function(layer, node) {
+  if (node.dimensions) {
+    var dimensions = {};
+    for (var key in node.dimensions) {
+      var value = this.dimensions[key];
+      if (value !== undefined) {
+        dimensions[key] = value;
+      }
+    }
+    if (!ol.object.isEmpty(dimensions)) {
+      var source = layer.getSource();
+      if (source instanceof ol.source.WMTS) {
+        source.updateDimensions(dimensions);
+      } else if (source instanceof ol.source.TileWMS || source instanceof ol.source.ImageWMS) {
+        source.updateParams(dimensions);
+      } else {
+        // the source is not ready yet
+        layer.once('change:source', function() {
```

Comment:
If `updateLayerDimensions_` is called multiple times before the `change:source` event is triggered. Won't it register many listeners?